### PR TITLE
Ns login mockup

### PR DESCRIPTION
**Pull request by natibekele**
1.  Create welcome screen view
2. Create User Creation View
3. Create Widgets for welcome and user creation views
      ProfileIcon, RecordButton, UsersList, WelcomeBack, Progressbar
4. Initialize folder/package layout structure

### CODE REVIEW

**Comment by natibekele** [comment]
Sorry for how long this is...

File: src/main/kotlin/app/ui/ImageLoader.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        else -> {
+            println("Error: Image file extension found is not svg, png, or jpg")
+            return ImageView(/*put a default image here*/)
+        }
+    }
+}
+
+
+//parameter example:
+//var imagePathToLoad = File("C:\\Users\\fucat\\Documents\\repositories\\8woc2018-jvm\\src\\main\\kotlin\\app\\ui\\micIcon-Copy.svg")
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/ImageLoader.kt]: let's clear out the commented code here

File: build.gradle [hunk 2]
```diff
@@ -13,7 +14,7 @@ plugins {
     id 'java'
 }
 
-group 'org.wycliffeassociates.translationrecorder'
+group 'com.kotlin.example'
```
Comment (jsarabia) [inline build.gradle]: please undo this change

File: src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+        val closeButton = button("CLOSE", mIcon) {
+            importStylesheet(ButtonStyles::class)
+            addClass(ButtonStyles.rectangleButtonDefault)
+
+            style {
+                alignment = Pos.CENTER
+                mIcon.fill = c("#CC4141")
+                effect = DropShadow(10.0, Color.GRAY)
+
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt]: let's clean up a lot of the unnecessary white space like you see above

File: src/main/kotlin/app/ui/styles/Styles.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package app.ui.styles
+
+import javafx.scene.Cursor
+import tornadofx.*
+
+class Styles: Stylesheet() {
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/styles/Styles.kt]: Styles of what?

Comment (natibekele) [reply src/main/kotlin/app/ui/styles/Styles.kt]: old file, not being used. I'll remove it

File: src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    var onClickRedo = PublishSubject.create<Boolean>()             // subject to check if the REDO button was clicked
+    var audioListened = PublishSubject.create<Boolean>()            // subject to check if the audio was listened
+
+    private val viewModel = ProfilePreviewViewModel(iconHash, onClickNext, onClickRedo, audioListened)
+
+    var NewUserButton = ProfileIcon("12345678901", 152.0)
+
+    val micIcon = MaterialIconView(MaterialIcon.MIC_NONE, "25px")
+    val rightArrow = MaterialIconView(MaterialIcon.ARROW_FORWARD, "25px")
+    val mIcon = MaterialIconView(MaterialIcon.CLOSE, "25px")
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt]: looks like this should be called "closeIcon"

File: src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+class ProfilePreview: View() {
+
+    var iconHash = PublishSubject.create<String>()                // subject to get the user iconHash
+    var onClickNext = PublishSubject.create<Boolean>()          // subject to check if the NEXT button was clicked
+    var onClickRedo = PublishSubject.create<Boolean>()             // subject to check if the REDO button was clicked
+    var audioListened = PublishSubject.create<Boolean>()            // subject to check if the audio was listened
+
+    private val viewModel = ProfilePreviewViewModel(iconHash, onClickNext, onClickRedo, audioListened)
+
+    var NewUserButton = ProfileIcon("12345678901", 152.0)
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt]: newUserButton

File: src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+                            action {
+
+                            }
+                        }
+
+
+                    }
+
+                    label("NEXT") {
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt]: use a properties file for strings (localizable)

File: src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+                            }
+                            action {
+                                viewModel.listenedAudio(false)
+                                find(ProfilePreview::class).replaceWith(UserCreation::class)
+
+                            }
+                        }
+                    }
+
+                    label("REDO") {
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt]: use a properties file for strings (localizable)

File: src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+                            style {
+                                backgroundColor += c("#CC4141")
+                                cursor = Cursor.HAND
+                                minWidth = 75.0.px
+                                minHeight = 75.0.px
+                                fontSize = 2.em
+                                textFill = c("#CC4141")
+                            }
+
+                            action {
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt]: don't have an action if there's nothing to put here?

File: src/main/kotlin/app/ui/profilePreview/ViewModel/ProfilePreviewViewModel.kt [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        private val userIconHash : PublishSubject<String>,
+        private val clickNext : PublishSubject<Boolean>,
+        private val clickRedo : PublishSubject<Boolean>,
+        private val userListenedAudio: PublishSubject<Boolean>
+
+) {
+
+    private val model = ProfilePreviewModel()
+
+    fun newIconHash( iconHash: String) {
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/profilePreview/ViewModel/ProfilePreviewViewModel.kt]: don't put a space before iconHash - same for other functions in this file

File: src/main/kotlin/app/ui/profilePreview/ViewModel/ProfilePreviewViewModel.kt [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+    fun clickNext( clicked: Boolean) {
+        clickNext.onNext(clicked)
+        model.clickNext = clicked
+    }
+
+    fun listenedAudio(listened: Boolean) {
+        userListenedAudio.onNext(listened)
+        model.listenedAudio = listened
+
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/profilePreview/ViewModel/ProfilePreviewViewModel.kt]: remove the empty line here

File: src/main/kotlin/app/ui/styles/ButtonStyles.kt [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+
+            rectangleButtonDefault {
+                fill = c("#CC4141")
+                backgroundColor+= c("#FFFF")
+                minWidth = 100.0.px
+                cursor= Cursor.HAND
+                effect = DropShadow(10.0, Color.GRAY)
+
+
+
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/styles/ButtonStyles.kt]: too much whitespace

File: src/main/kotlin/app/ui/userCreation/Model/UserCreationModel.kt [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+
+    var isRecording = false
+    lateinit var audioFile: String
+    lateinit var UserIcon: Image
+    var hasListened = false
+
+
+
+    fun recordClicked() {
+        if(isRecording == false) isRecording=true
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/userCreation/Model/UserCreationModel.kt]: space around the = (isRecording = true)

File: src/main/kotlin/app/ui/userCreation/View/UserCreation.kt [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+    }
+
+    private fun onClickCallback() {
+        viewModel.recordAudio()
+        viewModel.recordClicked()
+        timer.schedule(timerTask {
+            Platform.runLater {
+                viewModel.doneRecording()
+            }
+        }, 6100)
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/userCreation/View/UserCreation.kt]: set a constant at the top of the file for this, helps document the specific number

File: src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt (hunk 9 above)
Comment (natibekele) [reply src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt]: i believe it is there to 'doSomething' when the next button is clicked, but we don't have that screen yet. so i'll remove for now

File: src/main/kotlin/app/ui/userCreation/Model/UserCreationModel.kt [hunk 15]
```diff
@@ -0,0 +1,24 @@
+package app.ui.userCreation.Model
+
+import java.awt.Image
+
+class UserCreationModel {
+
+    var isRecording = false
+    lateinit var audioFile: String
+    lateinit var UserIcon: Image
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/userCreation/Model/UserCreationModel.kt]: userIcon

also, a nullable type is probably more appropriate here than a lateinit.

File: src/main/kotlin/app/ui/userCreation/View/UserCreation.kt [hunk 16]
```diff
@@ -0,0 +1,161 @@
...
+    private val mIcon = MaterialIconView(MaterialIcon.CLOSE, "25px")
+    private val viewModel : UserCreationViewModel  by inject ()
+    lateinit var recordButton: VBox
+    private val progressBar = ProgressBar()
+    val isRecording = viewModel.isRecording
+    val doneRecording = viewModel.doneRecording
+    var timer = Timer()
+
+    //initialize close button to be used in top of borderpane
+    val closeButton = button("CLOSE", mIcon) {
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/userCreation/View/UserCreation.kt]: properties file

File: src/main/kotlin/app/ui/userCreation/View/UserCreation.kt [hunk 17]
```diff
@@ -0,0 +1,161 @@
...
+                            }
+                        },
+
+                onError = {
+                            println(it)
+                        }
+                )
+
+            // record button needs to be added here below bc the subjects haven't emitted anything yet
+
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/userCreation/View/UserCreation.kt]: ...what?

Comment (jsarabia) [reply src/main/kotlin/app/ui/userCreation/View/UserCreation.kt]: OH that's because they should be PublishSubjects, not BehaviorSubjects. 

http://reactivex.io/RxJava/javadoc/rx/subjects/BehaviorSubject.html
http://reactivex.io/RxJava/javadoc/rx/subjects/PublishSubject.html
http://reactivex.io/RxJava/javadoc/rx/subjects/Subject.html

File: src/main/kotlin/app/ui/userCreation/View/UserCreation.kt [hunk 18]
```diff
@@ -0,0 +1,161 @@
...
+*  has two children. A close button, and a recordingFrame/recordButton that
+*  gives the user visual feedback through the recording process
+*
+* */
+class UserCreation : View("Creating User") {
+
+
+    private val mIcon = MaterialIconView(MaterialIcon.CLOSE, "25px")
+    private val viewModel : UserCreationViewModel  by inject ()
+    lateinit var recordButton: VBox
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/userCreation/View/UserCreation.kt]: don't lateinit this

File: src/main/kotlin/app/ui/userCreation/ViewModel/UserCreationViewModel.kt [hunk 19]
```diff
@@ -0,0 +1,86 @@
...
+    fun countdown() {
+
+        Platform.runLater { countdownTracker.set("3") }
+        setCountDown("2", 1)         // time in seconds
+        setCountDown("1", 2)
+        setCountDown("", 3)
+    }
+
+
+    private fun setCountDown(text: String, time: Int){
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/userCreation/ViewModel/UserCreationViewModel.kt]: time in what units?

File: src/main/kotlin/app/ui/welcomeScreen/View/WelcomeScreen.kt [hunk 20]
```diff
@@ -0,0 +1,106 @@
...
+        vbox {
+            alignment = Pos.CENTER
+            stackpane {
+                add(WelcomeBack());
+                style {
+                    setMinHeight(Window.HEIGHT.toDouble())
+                    backgroundColor += Color.WHITE
+                    vgrow = Priority.ALWAYS
+                    hgrow=Priority.ALWAYS
+                    //setMinWidth(Screen.getPrimary().bounds.width/2)
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/welcomeScreen/View/WelcomeScreen.kt]: remove this if you're not using it

File: src/main/kotlin/app/ui/welcomeScreen/View/WelcomeScreen.kt [hunk 21]
```diff
@@ -0,0 +1,106 @@
...
+        //put in some nice margins so it's not too crowded
+
+    }
+
+    //set the root of the view to the welcomeScreen
+    override val root = welcomeScreen;
+
+    //DON'T MOVE THIS TO THE TOP
+    //current window will be null unless init goes under setting of root
+    init{
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/welcomeScreen/View/WelcomeScreen.kt]: space after init

File: src/main/kotlin/app/widgets/recordButton/RecordingAnimation.kt [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+    lateinit var animation: Timeline
+
+    val bigCircle = circle{
+             centerX = 120.0
+             centerY = 120.0
+             radius = 120.0;
+             fill = c("#EDEDED");
+         }
+
+        val  arc = arc {
```
Comment (jsarabia) [inline src/main/kotlin/app/widgets/recordButton/RecordingAnimation.kt]: weird indentation

File: src/main/kotlin/app/ui/userCreation/View/UserCreation.kt (hunk 17 above)
Comment (natibekele) [reply src/main/kotlin/app/ui/userCreation/View/UserCreation.kt]: Ah ok. I thought that PublishSubjects behaved similarly because I had to do something similar in the rxdemo I was working on a couple weeks back. will go ahead and change it

File: src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt [hunk 23]
```diff
@@ -171,16 +164,11 @@ class ProfilePreview: View() {
                                 fontSize = 2.em
                                 textFill = c("#CC4141")
                             }
-
-                            action {
-
-                            }
+                            // TODO("insert action here when user clicks next button")
                         }
-
```
Comment (natibekele) [inline src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt]: I went in an removed a lot of the whitespace in this files. I feel how you see it all here doesn't really give an accurate representation of it though

File: src/main/kotlin/app/widgets/ProgressBar.kt [hunk 24]
```diff
@@ -0,0 +1,50 @@
...
+ * sweet code for a heckin cool loading bar
+ */
+class ProgressBar : VBox() {
+
+    init {
+
+        importStylesheet("/progressbar.css")
+        var messages = ResourceBundle.getBundle("MyView")
+
+        var isItDone = false
```
Comment (jsarabia) [inline src/main/kotlin/app/widgets/ProgressBar.kt]: because you update this value and check it in different threads, one of the threads may end up pulling a cached (dirty) value; as a result you need to have this field annotated as @Volatile

File: src/main/resources/UIColors_light.properties [hunk 25]
```diff
@@ -0,0 +1,5 @@
+white=#FFF
```
Comment (jsarabia) [inline src/main/resources/UIColors_light.properties]: Keep colors at 6 digit hex

File: src/main/kotlin/app/widgets/welcomeBack/WelcomeBack.kt [hunk 26]
```diff
@@ -0,0 +1,55 @@
...
+import javafx.scene.paint.Color
+import javafx.scene.text.FontWeight
+import tornadofx.*
+import app.widgets.profileIcon.ProfileIcon
+import java.awt.Button
+import java.util.*
+
+class WelcomeBack : HBox() {
+    private val rad = 125.0
+    val messages = ResourceBundle.getBundle("Messages")
```
Comment (jsarabia) [inline src/main/kotlin/app/widgets/welcomeBack/WelcomeBack.kt]: shouldn't tornadofx be doing this for you though?

File: src/main/kotlin/app/widgets/recordButton/RecordingAnimation.kt [hunk 27]
```diff
@@ -0,0 +1,70 @@
...
+import javafx.animation.Timeline
+import javafx.geometry.Pos
+import javafx.scene.layout.HBox
+import javafx.scene.paint.Color
+import javafx.scene.shape.ArcType
+import tornadofx.*;
+
+class RecordingAnimation : HBox() {
+
+    lateinit var animation: Timeline
```
Comment (jsarabia) [inline src/main/kotlin/app/widgets/recordButton/RecordingAnimation.kt]: I'd recommend using a nullable type here too

File: src/main/kotlin/app/widgets/recordButton/RecordButtonViewModel.kt [hunk 28]
```diff
@@ -0,0 +1,41 @@
...
+    fun countdown() {
+
+        Platform.runLater { countdownTracker.set("3") }
+        setCountDown("2", 1)         // time in seconds
+        setCountDown("1", 2)
+        setCountDown("", 3)
+
+    }
+
+    private fun setCountDown(text: String, time: Int){
```
Comment (jsarabia) [inline src/main/kotlin/app/widgets/recordButton/RecordButtonViewModel.kt]: if time is seconds, how about seconds for the variable name?

File: src/main/kotlin/app/widgets/recordButton/RecordButton.kt [hunk 29]
```diff
@@ -0,0 +1,86 @@
...
+
+
+import java.util.*
+import kotlin.concurrent.timerTask
+
+class RecordButton(var onClickCallback: () -> Unit = ::println, var animationCompletedCallback: () -> Unit = ::println, var stopClickedCallback: () -> Unit = ::println) : VBox() {
+
+
+    val circle = RecordingAnimation()
+    val dotsAn = DotsAnimation()
```
Comment (jsarabia) [inline src/main/kotlin/app/widgets/recordButton/RecordButton.kt]: I'd name this dotsAnim if you really need to abbreviate but dotsAnimation is probably better

File: src/main/kotlin/app/widgets/recordButton/RecordButton.kt [hunk 30]
```diff
@@ -0,0 +1,86 @@
...
+import tornadofx.*
+
+
+import java.util.*
+import kotlin.concurrent.timerTask
+
+class RecordButton(var onClickCallback: () -> Unit = ::println, var animationCompletedCallback: () -> Unit = ::println, var stopClickedCallback: () -> Unit = ::println) : VBox() {
+
+
+    val circle = RecordingAnimation()
```
Comment (jsarabia) [inline src/main/kotlin/app/widgets/recordButton/RecordButton.kt]: circleAnimation?

File: src/main/kotlin/app/widgets/welcomeBack/WelcomeBack.kt (hunk 26 above)
Comment (natibekele) [reply src/main/kotlin/app/widgets/welcomeBack/WelcomeBack.kt]: oh, inside the widgets I wasn't able to reference the bundled resources. i'm not sure as to why? I think it has to do with how/where the bundles are configured.

File: src/main/kotlin/app/ui/userCreation/View/UserCreation.kt [hunk 31]
```diff
@@ -0,0 +1,159 @@
...
+*  This View is used in the process of creating user profiles
+*  This View (UserCreation) Which serves as the parent view
+*  has two children. A close button, and a recordingFrame/recordButton that
+*  gives the user visual feedback through the recording process
+*
+* */
+class UserCreation : View() {
+
+
+    private val mIcon = MaterialIconView(MaterialIcon.CLOSE, "25px")
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/userCreation/View/UserCreation.kt]: closeIcon

File: src/main/kotlin/app/Main.kt [hunk 32]
```diff
@@ -0,0 +1,17 @@
+package app
+
+import app.ui.profilePreview.View.ProfilePreview
+import app.ui.welcomeScreen.*
+import tornadofx.*
+import java.util.*
+
+class MyApp : App(WelcomeScreen::class) {
+    companion object {
+        val Colors = ResourceBundle.getBundle("Colors")
```
Comment (jsarabia) [inline src/main/kotlin/app/Main.kt]: Can you see if this will work in it's own separate object? (specifically object, not class)

File: src/main/kotlin/app/ui/profilePreview/ViewModel/ProfilePreviewViewModel.kt [hunk 33]
```diff
@@ -0,0 +1,36 @@
...
+        clickRedo.onNext(clicked)
+        model.clickRedo = clicked
+    }
+
+    fun clickNext(clicked: Boolean) {
+        clickNext.onNext(clicked)
+        model.clickNext = clicked
+    }
+
+    fun listenedAudio(listened: Boolean) {
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/profilePreview/ViewModel/ProfilePreviewViewModel.kt]: this is a really awkward naming convention. I know what you're trying to communicate but it's not obvious at first. Not sure what yet, but let's try to think of a different way of phrasing this before committing it.

File: src/main/kotlin/app/ui/userCreation/View/UserCreation.kt [hunk 34]
```diff
@@ -0,0 +1,157 @@
...
+
+
+    private val closeIcon = MaterialIconView(MaterialIcon.CLOSE, "25px")
+    private val viewModel: UserCreationViewModel  by inject()
+    var recordButton = RecordButton(::onClickCallback, ::animationCompleted, ::stopClicked)
+    private val progressBar = ProgressBar()
+    val isRecording = viewModel.isRecording
+    val doneRecording = viewModel.doneRecording
+    var timer = Timer()
+    val RECORDING_DONE: Long = 6100
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/userCreation/View/UserCreation.kt]: This variable name makes it sound like a boolean value not a long. Perhaps RECORDING_TIME or TIME_ALLOTTED or something like that?

File: src/main/kotlin/app/ui/userCreation/View/UserCreation.kt [hunk 35]
```diff
@@ -0,0 +1,157 @@
...
+            isRecording.subscribeBy(
+                    onNext = {
+                        if (it == false) { // if isRecording set to false create new RecordButton()
+                            val newRecordButton = RecordButton(::onClickCallback, ::animationCompleted, ::stopClicked)
+                            newRecordButton.alignment = Pos.CENTER
+                            recordButton.replaceWith(newRecordButton)
+                            recordButton = newRecordButton
+                        }
+                    },
+
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/userCreation/View/UserCreation.kt]: don't have a new line in a parameter list

File: src/main/kotlin/app/ui/userCreation/View/UserCreation.kt [hunk 36]
```diff
@@ -0,0 +1,157 @@
...
+                    }
+            )
+
+            doneRecording.subscribeBy(
+                    onNext = {
+                        if (it == true) { //done recording = true? then navigate to profile preview
+                            timer.schedule(timerTask {
+                                Platform.runLater {
+                                    find(UserCreation::class).replaceWith(ProfilePreview::class,
+                                            transition = ViewTransition.Fade(0.3.seconds))
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/userCreation/View/UserCreation.kt]: line break styling

File: src/main/kotlin/app/widgets/recordButton/Record.kt [hunk 37]
```diff
@@ -0,0 +1,3 @@
+package app.widgets.recordButton
+
+data class Record(val audio: String, val image : String, val recording: Boolean)
```
Comment (jsarabia) [inline src/main/kotlin/app/widgets/recordButton/Record.kt]: RecordState perhaps?

File: src/main/kotlin/app/widgets/welcomeBack/WelcomeBack.kt [hunk 38]
```diff
@@ -0,0 +1,56 @@
...
+class WelcomeBack : HBox() {
+    private val rad = 125.0
+    private val bigIcons = borderpane() {
+
+        alignment = Pos.CENTER
+
+        style {
+            backgroundColor += c(Colors["base"])
+        }
+        val iconHash = ProfileIcon("12345678901", 150.0, true)
```
Comment (jsarabia) [inline src/main/kotlin/app/widgets/welcomeBack/WelcomeBack.kt]: Should profileIcon be a parameter to this widget? (or Node?) especially since we don't want to hardcode values in widgets

File: src/main/kotlin/app/widgets/welcomeBack/WelcomeBack.kt [hunk 39]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.*
+
+class WelcomeBack : HBox() {
+    private val rad = 125.0
+    private val bigIcons = borderpane() {
+
+        alignment = Pos.CENTER
+
+        style {
+            backgroundColor += c(Colors["base"])
```
Comment (jsarabia) [inline src/main/kotlin/app/widgets/welcomeBack/WelcomeBack.kt]: Widgets shouldn't be accessing app Colors

File: src/main/kotlin/app/widgets/recordButton/Record.kt (hunk 37 above)
Comment (natibekele) [reply src/main/kotlin/app/widgets/recordButton/Record.kt]: Actually not using this file.  I'll remove it

File: src/main/kotlin/app/Main.kt (hunk 32 above)
Comment (natibekele) [reply src/main/kotlin/app/Main.kt]: yes it does work. changed it

File: src/main/kotlin/app/ui/userCreation/View/UserCreation.kt (hunk 36 above)
Comment (jsarabia) [reply src/main/kotlin/app/ui/userCreation/View/UserCreation.kt]:
```
find(UserCreation::class)
    .replaceWith(
            ProfilePreview::class,
            transition = ViewTransition.Fade(0.3.seconds)
    )
````

File: src/main/kotlin/app/ui/profilePreview/ViewModel/ProfilePreviewViewModel.kt (hunk 33 above)
Comment (natibekele) [reply src/main/kotlin/app/ui/profilePreview/ViewModel/ProfilePreviewViewModel.kt]: userHasListened?

File: src/main/kotlin/app/ui/userCreation/View/UserCreation.kt (hunk 35 above)
Comment (natibekele) [reply src/main/kotlin/app/ui/userCreation/View/UserCreation.kt]: I'm unsure what you mean by this

File: src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt [hunk 40]
```diff
@@ -0,0 +1,167 @@
...
+import app.ui.welcomeScreen.WelcomeScreen
+import javafx.scene.paint.Color
+import tornadofx.*
+
+class ProfilePreview : View() {
+
+    var iconHash = PublishSubject.create<String>()                // subject to get the user iconHash
+    var onClickNext = PublishSubject.create<Boolean>()          // subject to check if the NEXT button was clicked
+    var onClickRedo = PublishSubject.create<Boolean>()             // subject to check if the REDO button was clicked
+    var audioListened = PublishSubject.create<Boolean>()            // subject to check if the audio was listened
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt]: seems like these things probably are better suited to belong to the viewmodel?

File: src/main/kotlin/app/ui/profilePreview/ViewModel/ProfilePreviewViewModel.kt [hunk 41]
```diff
@@ -0,0 +1,31 @@
+package app.ui.profilePreview.ViewModel
+
+import app.ui.profilePreview.Model.ProfilePreviewModel
+import io.reactivex.subjects.PublishSubject
+
+class ProfilePreviewViewModel(
+        private val userIconHash: PublishSubject<String>,
+        private val clickNext: PublishSubject<Boolean>,
+        private val clickRedo: PublishSubject<Boolean>,
+        private val userListenedAudio: PublishSubject<Boolean>) {
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/profilePreview/ViewModel/ProfilePreviewViewModel.kt]: ) { should go on a new line

File: src/main/kotlin/app/ui/userCreation/ViewModel/UserCreationViewModel.kt [hunk 42]
```diff
@@ -0,0 +1,56 @@
...
+        setCountDown("1", 2)
+        setCountDown("", 3)
+    }
+
+    private fun setCountDown(text: String, seconds: Int) {
+        timer.schedule(timerTask { Platform.runLater { countdownTracker.set(text) } }, seconds.toLong() * 1000)
+    }
+    fun stopCountdown() {
+        countdownTracker.set("")
+        timer?.cancel()
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/userCreation/ViewModel/UserCreationViewModel.kt]: timer is not a nullable?

File: src/main/kotlin/app/ui/welcomeScreen/View/WelcomeScreen.kt [hunk 43]
```diff
@@ -0,0 +1,98 @@
...
+                    }
+                }
+            }
+    }
+ }
+
+    //set the root of the view to the welcomeScreen
+    override val root = welcomeScreen
+
+    //DON'T MOVE THIS TO THE TOP
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/welcomeScreen/View/WelcomeScreen.kt]: can't you just declare root at the top as 
```
override val root = Hbox()
/*
other declarations
*/

init {
    //some init stuff

    with(root) {
        //layout stuff
    }
}
````

basically; seems like there's a cleaner way to do this?

File: src/main/kotlin/app/widgets/usersList/UsersList.kt [hunk 44]
```diff
@@ -0,0 +1,62 @@
...
+import javafx.geometry.Pos
+import javafx.scene.layout.HBox
+import javafx.scene.layout.Priority
+import app.ui.styles.ButtonStyles
+import tornadofx.*
+import app.widgets.profileIcon.ProfileIcon
+
+class UsersList : HBox() {
+
+    val data =listOf("11234567890", "12234567890", "12334567890", "12344567890", "12345567890", "02345667896", "12345677890", "02345678990", "12345678990")
```
Comment (jsarabia) [inline src/main/kotlin/app/widgets/usersList/UsersList.kt]: seems setting this needs to be exposed over some API

File: src/main/kotlin/app/widgets/ProgressBar.kt [hunk 45]
```diff
@@ -0,0 +1,42 @@
...
+import tornadofx.FX.Companion.messages
+import kotlin.concurrent.thread
+
+/**
+ * sweet code for a heckin cool loading bar
+ */
+class ProgressBar : VBox() {
+    init {
+        importStylesheet("/progressbar.css")
+        var isItDone = false
```
Comment (jsarabia) [inline src/main/kotlin/app/widgets/ProgressBar.kt]: needs to be volatile

File: src/main/kotlin/app/widgets/recordButton/RecordingAnimation.kt [hunk 46]
```diff
@@ -0,0 +1,60 @@
...
+import javafx.scene.shape.Arc
+import javafx.scene.shape.ArcType
+import javafx.scene.shape.Circle
+import tornadofx.*
+import java.time.Duration
+
+class RecordingAnimation : HBox() {
+    var animation: Timeline? = null
+    var arc: Arc? = null
+    var circle: Circle? = null
```
Comment (jsarabia) [inline src/main/kotlin/app/widgets/recordButton/RecordingAnimation.kt]: why are these nullable?

File: src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt [hunk 47]
```diff
@@ -0,0 +1,162 @@
...
+import tornadofx.*
+import app.widgets.WidgetsStyles
+
+class ProfilePreview : View() {
+
+    private val viewModel = ProfilePreviewViewModel()
+    var iconHash = viewModel.userIconHash
+    var onClickNext = viewModel.onClickNext
+    var onClickRedo = viewModel.onClickRedo
+    var audioListened =viewModel.audioListened
```
Comment (sarabiaj) [inline src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt]: these values are just going to end up out of sync with your viewmodel though?

File: src/main/kotlin/app/widgets/recordButton/RecordingAnimation.kt (hunk 46 above)
Comment (natibekele) [reply src/main/kotlin/app/widgets/recordButton/RecordingAnimation.kt]: they shouldn't be. I was trying to do some refactoring on that and should have changed it back.

File: src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt (hunk 47 above)
Comment (natibekele) [reply src/main/kotlin/app/ui/profilePreview/View/ProfilePreview.kt]: not if you inject the view model though right?
